datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -278,3 +278,71 @@ def _run_job_rate_limited(self):
 
     def run_job(self):
         raise NotImplementedError()
+
+
+def dbm_tracked_method(dbms, parent_check_attr=None, track_result_length=False):
```

Comment:
We can definitely use it more generally, let's move this to a new file `datadog_checks_base/datadog_checks/base/utils/tracking.py`
```suggestion
def tracked_method(integration, parent_check_attr=None, track_result_length=False):
```